### DISCUSSION TITLE
RPC compatibility with Lotus and Common Node API

### DISCUSSION THREAD

**Discussion by aatifsyed** [filecoin-project/venus#6381] [Ideas]
Hi Venus!

The [Forest](https://github.com/ChainSafe/forest) team are working on [a FIP for the Common Node API](https://github.com/filecoin-project/FIPs/pull/1027), with a machine-readable description of (a subset of) the existing RPC methods available for Lotus in [OpenRPC](https://spec.open-rpc.org/) format.

We wanted to engage you as implementers both of an RPC node, and as a potential future user of the document, with the following main queries:
- Does Venus strive to match Lotus's RPC API?
  - What's your current commitment to users r.e compatibility?
  - How do you ensure that you match currently?
  - Do you have any pain points as implementers?
- Do you have things that you'd want the OpenRPC document to address? 
  - Would you find the document useful?
  - What tooling would support you in your use of the OpenRPC document?

**Comment by simlecode** [comment]
* Venus API is compatible with Lotus (chain related)
  * The behavior of on-chain data-related API remains consistent.
  * Venus has tools to check whether the parameters of the lotus API are consistent and generate [documents](https://github.com/filecoin-project/venus/blob/master/venus-shared/compatible-checks/api-diff.txt).
  * Eth related interfaces are frequently updated.
* Can help users quickly understand the API functions and usage
  * The documentation is useful and can help new users quickly understand the specific functions and usage of the API.
  * None
